tests/bluechi_test/test.py

```diff
@@ -77,22 +83,41 @@ def setup(self) -> Tuple[bool, Tuple[BluechiControllerContainer, Dict[str, Bluec
                 node = BluechiNodeContainer(c, cfg)
                 node_container[cfg.node_name] = node
 
+                if self.run_with_valgrind:
+                    node.enable_valgrind()
+
                 node.exec_run('systemctl start bluechi-agent')
+                node.wait_for_unit_state_to_be("bluechi-agent.service", "active")
 
         except Exception as ex:
             success = False
             logger.debug(f"Failed to setup bluechi container: {ex}")
+            traceback.print_exc()
+
+        if self.run_with_valgrind:
+            # Give some more time for bluechi to start and connect while running with valgrind
+            time.sleep(2)
 
         return (success, (ctrl_container, node_container))
 
     def gather_logs(self, ctrl: BluechiControllerContainer, nodes: Dict[str, BluechiNodeContainer]):
         logger.debug("Collecting logs from all containers...")
+        valgrind_passed = True
 
         if ctrl is not None:
             ctrl.gather_journal_logs(self.tmt_test_data_dir)
+            if self.run_with_valgrind:
+                if not ctrl.gather_valgrind_logs(self.tmt_test_data_dir):
+                    valgrind_passed = False
 
         for _, node in nodes.items():
             node.gather_journal_logs(self.tmt_test_data_dir)
+            if self.run_with_valgrind:
+                if not node.gather_valgrind_logs(self.tmt_test_data_dir):
+                    valgrind_passed = False
+
+        if self.run_with_valgrind and not valgrind_passed:
+            raise Exception("Valgrind errors found in test. Review valgrind logs")
```

Comment:
We cannot raise exception here, because we will not gather test executor logs. Personally I'd only gather valgrind logs here same as for journal. And I'd add valgrind result evaluation into a special method executed after https://github.com/containers/bluechi/blob/main/tests/bluechi_test/test.py#L126 inside a try block (we don't need to evalutate valgrind if the test failed regularly)

Reply:
Good point, thanks.
I will restructure that

Reply:
Done, please review